### ISSUE TITLE
Unresolved var (in combination with defroutes?)

### ISSUE THREAD

**Issue by dharrigan** [clj-kondo/clj-kondo#1468]
**version**
`clj-kondo v2021.10.19`

**macro usage**

Configuration file is part of the project, in the `.clj-kondo` directory

**platform**

native version, but also ripples through from clojure-lsp

```
❯ uname -a                
Linux daffy 5.15.3-arch1-1 #1 SMP PREEMPT Thu, 18 Nov 2021 22:23:08 +0000 x86_64 GNU/Linux
```

**editor**

`neovim` with `conjure`

**problem**

I'm observing a problem in that `clj-kondo` is reporting `unused-var`, even 'tho it is used (and the config defines the defroutes macro). The problem **only** seems to manifest *if* there is a function defined before the `defroutes` macro.

If you comment out lines 6-9 in `a/routes.clj`, delete the `.clj-kondo/.cache` directory, then redo the lint check, all is fine.

**repro**

Please clone [this repo](https://github.com/dharrigan/unresolved-var.git). If you run `clj-kondo --lint src` in the root directory, you should observe this:

```
❯ clj-kondo --lint src                   
src/online/harrigan/api/router.clj:17:22: warning: Unresolved var: a-routes/public-routes
linting took 32ms, errors: 0, warnings: 1
```

**config**

Contained within the project.

**expected behavior**

It should not report unresolved-var.

Let me know if I can be of further assistance.

**Comment by borkdude** [comment]
@dharrigan Finally figured it out. The `:lint-as` configuration has to be on the top level, not under `:linters`.
